src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/RemoteProjectFileLoader.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.CodeAnalysis;
+using Roslyn.Utilities;
+using MSB = Microsoft.Build;
+
+namespace Microsoft.CodeAnalysis.MSBuild
+{
+    internal class RemoteProjectFileLoader
+    {
+        public static IProjectFileLoader CreateProjectLoader(Type loaderType)
+        {
+            return (IProjectFileLoader)GetBuildDomain().CreateInstanceAndUnwrap(loaderType.Assembly.FullName, loaderType.FullName);
+        }
+
+        // a separate app domain used for running msbuild
+        private static AppDomain s_BuildDomain;
+        private static object gate = new object();
+
+        private static AppDomain GetBuildDomain()
+        {
+            if (s_BuildDomain == null)
+            {
+                lock (gate)
+                {
+                    if (s_BuildDomain == null)
+                    {
+                        var thisAssembly = typeof(RemoteProjectFileLoader).Assembly.Location;
+                        var configFile = CreateTempFile(GetResourceText("msbuild.config"));
+
+                        s_BuildDomain = AppDomain.CreateDomain("MSBuildDomain",
+                            new System.Security.Policy.Evidence(),
+                            new AppDomainSetup
+                            {
+                                ApplicationBase = Path.GetDirectoryName(thisAssembly),
+                                ConfigurationFile = configFile
+                            });
+                    }
+                }
+            }
+
+            return s_BuildDomain;
+        }
+
+        private static string CreateTempFile(string content)
+        {
+            var filename = Path.GetTempFileName();
+            using (var stream = File.OpenWrite(filename))
+            {
+                using (var writer = new StreamWriter(stream, System.Text.Encoding.UTF8))
+                {
+                    writer.Write(content);
+                }
+            }
+
+            return filename;
+        }
+
+        private static string GetResourceText(string fileName)
```

Comment:
Can we not use a resource here? I worry it might get translated somehow, or at least give our localization team a heart attack with resources changing this late.